core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala

```diff
@@ -298,6 +300,10 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Opti
     }
 
     thrown.errorReport.statusCode shouldBe Some(StatusCodes.Conflict)
+
+    // Make sure that the pre-existing workspace was not deleted.
+    verify(workspaceManagerDAO, times(0)).deleteWorkspaceV2(any(), anyString(), any())
```

Comment:
I verified that both of these statements fail without the bug fix.

I considered also adding a test that ensures that a `RawlsExceptionWithErrorReport` with a different StatusCode still deletes the workspace, but looking through the code I didn't see any method we are calling in the creation block that would throw such an error. Do you think I should remove `RawlsExceptionWithErrorReport` from the deletion error handling all together?

Reply:
We need to keep `RawlsExceptionWithErrorReport` since that is thrown by `buildPolicyInputs` for invalid policies ([link](https://github.com/broadinstitute/rawls/blob/1824ef0ebf0834483542970b626f2841397bbdc0/model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala#L788)). It would be worth adding that other test to cover this IMO

Reply:
I don't know how removing `RawlsExceptionWithErrorReport` from the deletion error handling would work. What would you be matching on?

Reply:
I meant removing it from the exceptions that we handled and then deleted the workspace on (the following case statement). But Marc pointed out a reason why I should leave it there, so I will add a test case for that.

Reply:
@marctalbott actually, you already wrote a test for that case. :) https://github.com/broadinstitute/rawls/blob/1824ef0ebf0834483542970b626f2841397bbdc0/core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala#L646

I verified that the test case fails if I remove the handling for `RawlsExceptionWithErrorReport`.